litmus-portal/frontend/src/pages/MyHub/customMyHubCard.tsx

```diff
@@ -68,74 +73,83 @@ const CustomMyHubCard: React.FC<customMyHubCardProp> = ({
-                  className={classes.disconnectImg}
-                />
-                <Typography
-                  className={classes.disconnectText}
-                  data-cy="viewHub"
+          {userData.userRole !== 'Viewer' ? (
+            <>
+              <IconButton
+                aria-label="more"
+                aria-controls="long-menu"
+                aria-haspopup="true"
+                onClick={handleClick}
+                data-cy="browseScheduleOptions"
+                className={classes.iconButton}
+              >
+                <MoreVertIcon className={classes.cardOption} />
+              </IconButton>
+              <Menu
+                id="long-menu"
+                anchorEl={anchorEl}
+                keepMounted
+                open={open}
+                onClose={handleClose}
+              >
+                <MenuItem
+                  value="Refresh"
+                  onClick={() => {
+                    handleRefresh(hub.id);
+                  }}
+                >
+                  <div className={classes.cardMenu}>
+                    <img
+                      src="./icons/refresh.svg"
+                      alt="Refresh"
+                      className={classes.refreshImg}
+                    />
+                    <Typography data-cy="viewHub">
+                      {t('myhub.refresh')}
+                    </Typography>
+                  </div>
+                </MenuItem>
+                <MenuItem
+                  value="View"
+                  onClick={() => {
+                    history.push(`/myhub/edit/${hub.HubName}`);
+                  }}
+                >
+                  <div className={classes.cardMenu}>
+                    <img
+                      src="./icons/Edit.svg"
+                      alt="Edit"
+                      className={classes.editImg}
+                    />
+                    <Typography data-cy="viewHub">{t('myhub.edit')}</Typography>
+                  </div>
+                </MenuItem>
+
+                <MenuItem
+                  value="Delete"
+                  onClick={() => {
+                    handleDelete(hub.id);
+                  }}
                 >
-                  {t('myhub.disconnect')}
-                </Typography>
-              </div>
-            </MenuItem>
-          </Menu>
+                  <div className={classes.cardMenu}>
+                    <img
+                      src="./icons/bin-red.svg"
+                      alt="disconnect"
+                      className={classes.disconnectImg}
+                    />
+                    <Typography
+                      className={classes.disconnectText}
+                      data-cy="viewHub"
+                    >
+                      {t('myhub.disconnect')}
+                    </Typography>
+                  </div>
+                </MenuItem>
+              </Menu>
+            </>
+          ) : (
+            <></>
```

Comment:
A small suggestion, we can reduce these 2 lines by chaining with the && operator as there is nothing to render in the false case

Reply:
Thanks @vanshBhatia-A4k9 . I have updated the changes.